### ISSUE TITLE
Using both 'max-resources' AND 'max-resources-percent' together in policy validation

### ISSUE THREAD

**Issue by bb1314** [cloud-custodian/cloud-custodian#3712]
Hi there! For policy validation, what would happen if I use both 'max-resources' and 'max-resources-percent' together? I assume if either one condition is met, policy validation would fail? Is it possible such that both of those conditions have to be met in order for policy validation to fail? I'm running into use cases where I'd need to do both.

Ex:
max-resources: 2 AND max-resources-percent: 50%
max-resources: 2 OR max-resources-percent: 50%

**Comment by kapilt** [comment]
do you have a more worked real world example of needing this differential behavior/configuration.  right now the behavior is `or` and there are no plans to change that.

**Comment by kapilt** [comment]
also clarifying terms, these configuration values are not part of validation, they are a runtime execution circuit breaker. validation takes place statically pre execution and ANY api calls.

**Comment by bb1314** [comment]
Thanks for the quick response and clarification! We have an organizational wide policy to add circuit breaker logic all custodian policies that have action that we consider destructive (delete or update). The problem with that is that we decided that the default 'max-resources-percent' would be say 10% across the board, but certain types of resources by nature wouldn't have a very large numbers (but not guaranteed to always be a small number) in an account (eg: ECR, S3) and usually wouldn't hit that 10% (1 out of like 3 is already 33.33333%), in the case having AND logic makes sense so that I can do max-resources: 2 AND max-resources-percent: 10% for a delete action.

**Comment by anovis** [comment]
would something like this work. so max-resource can take in a dict in addition to an int.

```policies:
  - name: my-first-policy
    resource: sns
    max-resources: 
      - percent: 20
      - amount: 5
```

**Comment by kapilt** [comment]
Thanks i like that direction, although it doesn't actual consider/address the feature in question (ie. at the moment you can already specify both of those on a policy) it does perhaps suggest a policy configuration mechanism for them. The reality is those that i'm not sure if we should  expose operator policy configuration because the use case is typically uniform across the set of policies, or let me rephrase we can expose but also look at a blanket option.  ie. I was  considering an env var flag for max resources operator (and|or) that could configure across set of policies being executed..

**Comment by anovis** [comment]
ah ok so like below. 

```
  - max-resource: 5
  - max-resource-percent: 20
  - max-resource-op: and
  - name: my-first-policy
    resource: sns
    - ...
```

**Comment by kapilt** [comment]
i prefer your syntax of encapsulating them under a single attribute to avoid policy to level attr pollution

**Comment by anovis** [comment]
awesome. will create a pr.
